File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/Java/src/Model/Polygons/Square.java
package Model.Polygons;

import Model.Point;

import java.util.List;

public class Square extends Rectangle {
    public Square(final List<Point> vertices) throws Exception {
        super(vertices);
    }

    /**
     * Returns whether a polygon is a square or not
     */
    public static boolean canBeCreatedWith(final Polygon polygon) throws Exception {
        return Rectangle.canBeCreatedWith(polygon) && Rhombus.canBeCreatedWith(polygon);
    }

    @Override
    public double getApothemConstant() {
        return 0.5;
    }

    @Override
    public double getArea() throws Exception {
        return Math.pow(getBase(), 2);
    }

    @Override
    public double getPerimeter() throws Exception{
        return getBase() * 4;
    }

    @Override
    public String toString() {
        return "Square";
    }
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/C++/Model/Polygons/Square.cpp
#include "Square.h"

Square::Square(const list<Point> &vertices) : Quadrilateral(vertices), Parallelogram(vertices), Rectangle(vertices),
                                              Rhombus(vertices) {}

bool Square::canBeCreatedWith(const Polygon &polygon) {
    return Rectangle::canBeCreatedWith(polygon) && Rhombus::canBeCreatedWith(polygon);
}

double Square::getApothemConstant() const {
    return 0.5;
}

double Square::getArea() const {
    return pow(getBase(), 2);
}

double Square::getPerimeter() const {
    return Rhombus::getPerimeter();
}

string Square::getName() const {
    return "Square";
}